lib/src/git_backend.rs

```diff
@@ -190,14 +190,23 @@ impl GitBackend {
         )
         .map_err(GitBackendInitError::InitRepository)?;
         let git_repo_path = workspace_root.join(".git");
-        Self::init_with_repo(settings, store_path, &git_repo_path, git_repo)
+        Self::init_with_repo(store_path, &git_repo_path, git_repo, true)
     }
 
     /// Initializes backend with an existing Git repo at the specified path.
     pub fn init_external(
         settings: &UserSettings,
         store_path: &Path,
         git_repo_path: &Path,
+    ) -> Result<Self, Box<GitBackendInitError>> {
+        Self::init_external_with_tree_conflicts(settings, store_path, git_repo_path, true)
+    }
+
+    fn init_external_with_tree_conflicts(
+        settings: &UserSettings,
+        store_path: &Path,
+        git_repo_path: &Path,
+        imported_commit_uses_tree_conflict_format: bool,
```

Comment:
nit: maybe we can instead extract `git_backend.import_head_commits_with_tree_conflicts(.., uses_tree_conflict_format: bool)` for tests.

(can be addressed separately)

Reply:
Ah, that's much cleaner, thanks! Done in this PR.